src/parser/here_doc.c

```diff
@@ -39,37 +40,53 @@ static t_exit_code	append_line_to_heredoc(char *line,
 	return (SUCCESS);
 }
 
-static void	cleanup_here_doc(char *line, int fd)
+static int	cleanup_here_doc(char *line, int fd, char *file_name)
 {
+	int	*signal;
+	int	reopen_fd;
+
+	signal = heredoc_sigint();
 	close(fd);
 	free(line);
+	if (*signal)
+	{
+		*signal = 0;
+		reopen_fd = FILE_ERROR;
+	}
+	else
+		reopen_fd = open(file_name, O_RDONLY, FILE_RIGHTS);
+	if (reopen_fd == INVALID_FD)
+		handle_errors(19, "here_doc");
+	return (reopen_fd);
 }
 
 /*
 	syntax checker responsible for assuring that this is only entered
-	when ere is a valid here_doc token
+	when there is a valid here_doc token
 */
 int	handle_here_doc(const char *delimeter)
 {
 	char		file_name[BUFFER_SIZE];
 	int			fd;
 	char		*line;
+	int			*signal;
 
 	get_file_name(&file_name[0], delimeter);
 	fd = open(file_name, O_RDWR | O_CREAT | O_APPEND, FILE_RIGHTS);
 	if (fd == INVALID_FD)
 		handle_errors(19, "here_doc");
 	while (TRUE)
 	{
+		set_heredoc_signals();
 		line = readline("> ");
-		if (append_line_to_heredoc(line, delimeter, fd) == ERROR)
+		signal = heredoc_sigint();
+		if (*signal == 666)
```

Comment:
I think 666 isn't a good number..

Reply:
hehe I'm actually using 666 for something in heredoc already, so, true hehe :D